projects/plugins/jetpack/tests/php/core-api/wpcom-endpoints/test-class-wpcom-rest-api-v2-endpoint-admin-menu.php

```diff
@@ -252,6 +252,44 @@ public function submenu_item_data() {
 		);
 	}
 
+	/**
+	 * Check if the menu URL is properly generated from the first submenu slug.
+	 */
+	public function test_if_the_first_submenu_url_is_used_for_menu_url() {
+		global $menu;
+
+		add_menu_page( '', __( 'Foo', 'jetpack' ), 'read', 'foo' );
+		$fnc = function () { }; /// needed for the slug to register as a page.
+		add_submenu_page( 'foo', 'title', 'title', 'read', 'sharing', $fnc, 0 );
+
+		$foo_item = array();
+
+		foreach ( $menu as $menu_item ) {
+			if ( 'foo' === $menu_item[2] ) {
+				$foo_item = $menu_item;
+				break;
+			}
+		}
+
+		$class = new ReflectionClass( 'WPCOM_REST_API_V2_Endpoint_Admin_Menu' );
+
+		$prepare_menu_item = $class->getMethod( 'prepare_menu_item' );
+		$prepare_menu_item->setAccessible( true );
```

Comment:
What if we change the signature of the `prepare_menu_item ` method so it's `public` to avoid this? I don't think we have any good reason for keeping it private.

Reply:
Personally I prefer having them private, even though that means testing would be a bit harder, because we can differentiate between which methods are endpoints (and are part of the public API) and which are just helper methods. 

But if we'll make this public, then we should also make prepare_submenu_item public? :)